.concourse/product-release/pipeline.template

```diff
@@ -0,0 +1,467 @@
+            make kubeconfig
+` }}
+
+{{- $import_eks := `
+            # create builfolder prepared for eks
+            export BACKEND=eks
+            export AWS_ACCESS_KEY_ID='((aws-ci-chuller-access-key-id))'
+            export AWS_SECRET_ACCESS_KEY='((aws-ci-chuller-secret-access-key))'
+            export KUBECFG=$PWD/kubeconfig_$CLUSTER_NAME
+            make kubeconfig
+` }}
+
+{{- $import_aks := `
+            # create builfolder prepared for aks
+            export BACKEND=aks
+            export KUBECFG=$PWD/kubeconfig_$CLUSTER_NAME
+            make kubeconfig
+` }}
+
+{{- $import_caasp4 := `
+            # create builfolder prepared for caasp4os
+            export BACKEND=caasp4os
+            export KUBECFG=$PWD/kubeconfig_$CLUSTER_NAME
+            make kubeconfig
+` }}
+
+{{- $deploy := `
+            # deploy kubecf from chart
+            export SCF_CHART="$(readlink -f ../s3.kubecf-bundle/*.tgz)"
+            export SCF_TESTGROUP=true
+            export SCF_OPERATOR=true
+            export DOCKER_ORG=cap-staging
+            export QUIET_OUTPUT=true
+            make scf
+` }}
+
+{{- $test := `
+            # run test suites on cluster
+            # See: https://github.com/SUSE/catapult/wiki/Running-SCF-tests#kubecf
+            export KUBECF_TEST_SUITE="${TEST_SUITE:-smokes}"
+            export KUBECF_NAMESPACE="scf"
+            export SCF_TESTGROUP=true
+            export QUIET_OUTPUT=true
+            make tests-kubecf
+` }}
+
+{{- $stratos := `
+            # deploy stratos console & metrics
+            export QUIET_OUTPUT=true
+            make stratos metrics
+` }}
+
+{{- $upgrade := `
+            # upgrade kubecf
+            export SCF_CHART="$(readlink -f ../s3.kubecf-bundle/*.tgz)"
+            export SCF_OPERATOR=true
+            export QUIET_OUTPUT=true
+            make scf-chart # this takes the same chart as the deploy step for now
+            make scf-gen-config
+            make scf-upgrade
+` }}
+
+{{- $allbells := `
+        # enable as much options as possible
+        HA: true
+        EMBEDDED_UAA: true
+        AUTOSCALER: true
+` }}
+
+jobs:
+{{ range $_, $Backend := (ds "BACKEND").backend -}} # for all backends
+
+{{ range $_, $OptionsFlag := (ds "OPTIONS").options -}} # for options
+
+{{ range $_, $EiriniFlag := (ds "EIRINI").eirini -}} # for eirini
+
+# {{$Backend}}-{{$OptionsFlag}}
+
+- name: deploy-{{$EiriniFlag}}-{{$Backend}}-{{$OptionsFlag}}
+  public: false
+  plan:
+  - get: s3.kubecf-bundle
+    trigger: true
+  - get: kubecf-master
+  - get: catapult
+  - task: deploy
+    privileged: true
+    timeout: 2h30m
+    input_mapping:
+      kubecf: kubecf-master
+    config:
+      platform: linux
+      image_resource:
+        type: registry-image
+        source:
+          repository: splatform/catapult
+      inputs:
+      - name: kubecf
+      - name: catapult
+      - name: s3.kubecf-bundle
+      outputs:
+      - name: output
+      params:
```

Comment:
If you have a template, you can reduce quite a bit the duplication for the params. Take the example below (not tested, and you will also need to take some of the variables via the template arguments):

```yaml
{{- define "params" }}
AUTOSCALER: {{ eq $OptionsFlag "all" }}
CLUSTER_NAME: {{ $pipelineName }}-{{ $EiriniFlag }}-{{ $Backend }}-{{ $OptionsFlag }}
DEFAULT_STACK: cflinuxfs3
EKCP_HOST: ((ekcp-host))
ENABLE_EIRINI: {{ or (eq $EiriniFlag "eirini") (eq $OptionsFlag "all") }}
HA: {{ eq $OptionsFlag "ha" }}
QUIET_OUTPUT: true
{{- end }}
```

This is how you would render the params (you will need to pass the variables to the template):

```suggestion
      params:
        {{- tmpl.Exec "params" | nindent 8 }}
```